BendAssist.App/BendAssists/BendDeduction.cs

```diff
@@ -11,20 +13,157 @@ public sealed class BendDeduction : BendAssist {
+   static int[] GetCPIndices (PLine refPLine, List<PLine> pLines) {
+      var (start, end) = (refPLine.StartPoint, refPLine.EndPoint);
+      return pLines.Where (c => c.Index != refPLine.Index && (c.HasVertex (start) || c.HasVertex (end))).Select (c => c.Index).ToArray ();
+   }
    #endregion
 
    #region Implementation -------------------------------------------
+   /// <summary>Returns the list of bend lines after translating them</summary>
+   // Each bendLine is moved towards the base by a sum of half of its bend deduction and 
+   // the total of the bend deduction of all the inner bendLines
+   List<BendLine> GetTranslatedBLines (List<BendLine> bLines, out double totalBD, out int horBLCount, out int verBLCount, bool isNegOff = false) {
+      var newBendLines = new List<BendLine> ();
+      var offFactor = isNegOff ? -1 : 1;
+      totalBD = 0.0;
+      horBLCount = verBLCount = 0;
+      foreach (var bl in bLines) {
+         var (bd, orient) = (bl.BLInfo.Deduction, bl.Orientation);
+         var offset = offFactor * (totalBD + 0.5 * bd);
+         (double dx, double dy) = (0, 0);
+         if (orient is Horizontal) {
+            horBLCount += 1;
+            (dx, dy) = (0.0, offset);
+         } else if (orient is Vertical) {
+            verBLCount += 1;
+            (dx, dy) = (offset, 0.0);
+         }
+         totalBD += bd;
+         newBendLines.Add ((BendLine)bl.Translated (dx, dy));
+      }
+      return newBendLines;
+   }
+
+   /// <summary>Returns the edges parallel to the bend lines</summary>
+   List<PLine> GetProfilePLines (Part part, EBLLocation loc, bool HasVerBLine, bool HasHorBLine) {
+      var b = part.Bound;
+      List<PLine> alignedPLines = [];
+      if (HasVerBLine) { // Handles vertical bend lines and returns nearest vertical pLine 
+         alignedPLines.Add (part.PLines.Where (c => loc is EBLLocation.Top ? c.StartPoint.X == b.MaxX && c.EndPoint.X == b.MaxX
+                                                                         : c.StartPoint.X == b.MinX && c.EndPoint.X == b.MinX).First ());
+      }
+
+      if (HasHorBLine) { // Handles horizontal bend lines and returns nearest horizontal pLine 
+         alignedPLines.Insert (0, part.PLines.Where (c => loc is EBLLocation.Top ? c.StartPoint.Y == b.MaxY && c.EndPoint.Y == b.MaxY
+                                                                               : c.StartPoint.Y == b.MinY && c.EndPoint.Y == b.MinY).First ());
+      }
+      return alignedPLines;
+   }
    #endregion
 
    #region Private Data ---------------------------------------------
-   readonly EBDAlgorithm mAlgorithm;
+   readonly EBDAlgorithm mAlgorithm; // Algorithm by which bend deduction is applied
    #endregion
 }
+#endregion
+
+// TO be moved to Model - BEnum.cs
+#region Enums -------------------------------------------------------------------------------------
+public enum EBLLocation { Top, Bottom }

```

Comment:
Change EBLLocation to ELoc and add the following entries Bottom, Left, Right, Top